src/dag.c

```diff
@@ -818,7 +842,8 @@ int RedisAI_DagRunSyntaxParser(RedisModuleCtx *ctx, RedisModuleString **argv,
       } 
       int *instance = AI_dictGetVal(mangled_entry);
       RedisModuleString *mangled_key = RedisModule_CreateStringPrintf(ctx, "%s%04d", key, *instance);
-      AI_dictAdd(mangled_persisted, (void *)RedisModule_StringPtrLen(mangled_key, NULL), (void *)1);
+      const char* mangled_key_str = RedisModule_Strdup(RedisModule_StringPtrLen(mangled_key, NULL));
```

Comment:
Remove Strdup

Reply:
Missed that one :) fixed in the latest commit